File: Practice/06/C++/example6/example6/example6.cpp
#include <iostream>
#include <math.h>
using namespace std;

int main()
{
    setlocale(LC_ALL, "Russian");


    double a, b, c, D,x1,x2;

    cin >> a;
    cin >> b;
    cin >> c;
    
        D = b * b - 4 * a * c;

    if (D > 0 && a != 0 && b != 0 && c != 0) {
        x1 = (-b + sqrt(D)) /(2*a);
        x2 = (-b - sqrt(D)) /(2*a);
        cout << "x1 = " << x1 << "\n" << "x2 = " << x2;
    }else if (D == 0 && a != 0 && b != 0 && c != 0) {
        x1 = -b / (2 * a);
        cout << "x1 = " << x1;
    }else if (D < 0 && a != 0 && b != 0 && c != 0) {
        cout << "Нет корней";


    }else if (a == 0 && b != 0 && c != 0) {
        x1 = -c / b;
        cout << "x1 = " << x1;
    }else if (a != 0 && b == 0 && c != 0) {
                if (-c / a >= 0) {
                    x1 = +sqrt(-c / a);
                    x2 = -sqrt(-c / a);
                    cout << "x1 = " << x1 << "\n" << "x2 = " << x2;
                }
                else {
                    cout << "Нет корней";
                }

    }else if (a != 0 && b != 0 && c == 0) {
        x1 = 0;
        x2 = -b / a;
        cout << "x1 = " << x1 << "\n" << "x2 = " << x2;

    }else if (a == 0 && b == 0 && c == 0) {
        cout << "x = R";
    }else if (a == 0 && b != 0 && c == 0) {
            x1 = 0;
            cout << "x1 = " << x1;
    }else if (a == 0 && b == 0 && c != 0) {
        if (c == 0) {
            cout << "x = R";
        }
        else {
            cout << "Нет корней";// c != 0
        }

    }else if (a != 0 && b == 0 && c == 0) {
        x1 = 0;
        cout << "x1 = " << x1;
    }






}

File: Practice/06/Python/example6.py
import math

a = int(input())
b = int(input())
c = int(input())


D = b*b - 4*a*c


if D > 0 and a != 0 and b != 0 and c != 0:
    x1 = (-b + math.sqrt(D)) /(2*a)
    x2 = (-b - math.sqrt(D)) /(2*a)
    print("x1 = " + str(x1) + "\nx2 = " + str(x2))
elif (D == 0 and a != 0 and b != 0 and c != 0):
    x1 = -b / (2 * a)
    print("x1 = " + str(x1))
elif (D < 0 and a != 0 and b != 0 and c != 0):
    print("Нет корней")
elif (a == 0 and b != 0 and c != 0):
    x1 = -c / b
    print("x1 = " + str(x1))
elif (a != 0 and b == 0 and c != 0):
	if (-c / a >= 0):
		x1 = +math.sqrt(-c / a)
		x2 = -math.sqrt(-c / a)
		print("x1 = " + str(x1) + "\nx2 = " + str(x2))
	else:
		print("Нет корней")
elif (a != 0 and b != 0 and c == 0):

    x1 = 0
    x2 = -b / a
    print("x1 = " + str(x1) + "\nx2 = " + str(x2))
elif (a == 0 and b == 0 and c == 0):
    print("x = R")
elif (a == 0 and b != 0 and c == 0):
    x1 = 0
    print( "x1 = " + str(x1))
elif (a == 0 and b == 0 and c != 0):
    if (c == 0):
        print("x = R")
    else:
        print("Нет корней")# c != 0
elif (a != 0 and b == 0 and c == 0):
    x1 = 0
    print( "x1 = " + str(x1))
    


